src/main/java/com/eternalcode/combat/fight/FightTask.java

```diff
@@ -31,8 +33,11 @@ public FightTask(Server server, PluginConfig config, FightManager fightManager,
     }
 
     @Override
+
+
     public void run() {
-        for (FightTag fightTag : this.fightManager.getFights()) {
+        List<FightTag> fights = new ArrayList<>(this.fightManager.getFights());
```

Comment:
It won't be efficient. 
To avoid ConcurrentModificationException, I suggest to use ConcurrentHashMap in FightManager.